File: examples/modulith-clinical-tool-jpa/src/test/java/io/zenwave360/example/clinicaltool/modules/clinical/config/RepositoriesInMemoryConfig.java
package io.zenwave360.example.clinicaltool.modules.clinical.config;

import io.zenwave360.example.clinicaltool.modules.clinical.core.outbound.jpa.DoctorRepository;
import io.zenwave360.example.clinicaltool.modules.clinical.core.outbound.jpa.HospitalRepository;
import io.zenwave360.example.clinicaltool.modules.clinical.core.outbound.jpa.PatientRepository;
import io.zenwave360.example.clinicaltool.modules.clinical.infrastructure.jpa.inmemory.DoctorRepositoryInMemory;
import io.zenwave360.example.clinicaltool.modules.clinical.infrastructure.jpa.inmemory.HospitalRepositoryInMemory;
import io.zenwave360.example.clinicaltool.modules.clinical.infrastructure.jpa.inmemory.PatientRepositoryInMemory;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Primary;

// @Configuration
public class RepositoriesInMemoryConfig {

    protected final HospitalRepository hospitalRepository = new HospitalRepositoryInMemory();

    @Bean
    @Primary
    public <T extends HospitalRepository> T hospitalRepository() {
        return (T) hospitalRepository;
    }

    protected final DoctorRepository doctorRepository = new DoctorRepositoryInMemory();

    @Bean
    @Primary
    public <T extends DoctorRepository> T doctorRepository() {
        return (T) doctorRepository;
    }

    protected final PatientRepository patientRepository = new PatientRepositoryInMemory();

    @Bean
    @Primary
    public <T extends PatientRepository> T patientRepository() {
        return (T) patientRepository;
    }
}

File: examples/modulith-klinical-tool-jpa/src/test/kotlin/io/zenwave360/example/clinicaltool/modules/clinical/config/RepositoriesInMemoryConfig.kt
package io.zenwave360.example.clinicaltool.modules.clinical.config

import io.zenwave360.example.clinicaltool.modules.clinical.core.outbound.jpa.*
import io.zenwave360.example.clinicaltool.modules.clinical.infrastructure.jpa.inmemory.*
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Primary

// @Configuration
open class RepositoriesInMemoryConfig {

    protected val hospitalRepository = HospitalRepositoryInMemory()

    @Bean
    @Primary
    fun hospitalRepository(): HospitalRepositoryInMemory {
        return hospitalRepository
    }

    protected val doctorRepository = DoctorRepositoryInMemory()

    @Bean
    @Primary
    fun doctorRepository(): DoctorRepositoryInMemory {
        return doctorRepository
    }

    protected val patientRepository = PatientRepositoryInMemory()

    @Bean
    @Primary
    fun patientRepository(): PatientRepositoryInMemory {
        return patientRepository
    }
}
